drivers/watchdog/wdt_opentitan.c

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2023 by Rivos Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+#include <soc.h>
+#include <zephyr/drivers/watchdog.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(wdt_opentitan, CONFIG_WDT_LOG_LEVEL);
+
+#define DT_DRV_COMPAT lowrisc_opentitan_aontimer
+
+#define REG_OT_WDT(dev) \
+	(((const struct wdt_ot_aontimer_cfg *const)(dev)->config)->regs)
+#define NODE_AON DT_NODELABEL(aontimer)
+
+struct ot_aontimer_regs {
+	uint32_t alert_test;
+	uint32_t wkup_ctrl;
+	uint32_t wkup_thold;
+	uint32_t wkup_count;
+	uint32_t wdog_regwen;
+	uint32_t wdog_ctrl;
+	uint32_t wdog_bark_thold;
+	uint32_t wdog_bite_thold;
+	uint32_t wdog_count;
+	uint32_t intr_state;
+} __packed;
+
+struct wdt_ot_aontimer_cfg {
+	volatile struct ot_aontimer_regs *regs;
+	uint32_t clk_freq;
+	bool wdog_lock;
+};
+
+struct wdt_ot_aontimer_data {
+	uint8_t reset_type;
+	wdt_callback_t bark;
+	bool en;
+};
+
+
+static int ot_aontimer_setup(const struct device *dev, uint8_t options)
+{
+	const struct wdt_ot_aontimer_cfg *const cfg = dev->config;
+	volatile struct ot_aontimer_regs *regs = REG_OT_WDT(dev);
+
+	regs->wdog_count = 0;
+	regs->wdog_ctrl |= BIT(0);
```

Comment:
In general it is preferred to use the `sys_write` / `sys_read` helpers to access and work with MMIO registers. This is done because usually the helpers are also using memory barriers to ensure correctness.